IndexingTest_SpectrumNumberTranslatorTest failing on Ubuntu 14.04

`IndexingTest_SpectrumNumberTranslatorTest` has been failing on Ubuntu 14.04 for a while http://builds.mantidproject.org/job/master_clean-ubuntu/

```
test 179
    Start 179: IndexingTest_SpectrumNumberTranslatorTest

179: Test command: /usr/bin/cmake "-E" "chdir" "/home/builder/jenkins-linode/workspace/master_clean-ubuntu/build/bin/Testing" "/home/builder/jenkins-linode/workspace/master_clean-ubuntu/build/bin/IndexingTest" "SpectrumNumberTranslatorTest"
179: Test timeout computed to be: 300
179: Running 23 tests...
179: In SpectrumNumberTranslatorTest::test_spectrum_numbers_order_preserved:
179: /home/builder/jenkins-linode/workspace/master_clean-ubuntu/Framework/Indexing/test/SpectrumNumberTranslatorTest.h:116: Error: Test failed: Unknown error -1
179: /home/builder/jenkins-linode/workspace/master_clean-ubuntu/Framework/Indexing/test/SpectrumNumberTranslatorTest.h:117: Error: Test failed: Unknown error -1
179: /home/builder/jenkins-linode/workspace/master_clean-ubuntu/Framework/Indexing/test/SpectrumNumberTranslatorTest.h:118: Error: Test failed: Unknown error -1
179: /home/builder/jenkins-linode/workspace/master_clean-ubuntu/Framework/Indexing/test/SpectrumNumberTranslatorTest.h:119: Error: Test failed: Unknown error -1
179: ......terminate called after throwing an instance of 'std::system_error'
179:   what():  Unknown error -1
179: Child aborted
1/1 Test #179: IndexingTest_SpectrumNumberTranslatorTest ...***Failed    0.12 sec

0% tests passed, 1 tests failed out of 1
```

--- a/Framework/Indexing/CMakeLists.txt
+++ b/Framework/Indexing/CMakeLists.txt
@@ -69,6 +69,12 @@ set_property ( TARGET Indexing PROPERTY FOLDER "MantidFramework" )
 
 target_link_libraries ( Indexing LINK_PRIVATE ${TCMALLOC_LIBRARIES_LINKTIME} ${MANTIDLIBS} )
 
+if( "${UNIX_CODENAME}" STREQUAL "trusty" )
+  # Special linking options are required for some older Ubuntu gcc builds to use std::thread, specific options found here:
+  # http://stackoverflow.com/questions/8649828/what-is-the-correct-link-options-to-use-stdthread-in-gcc-under-linux#comment35723468_8649908
+  target_link_libraries ( Indexing LINK_PRIVATE -Wl,--no-as-needed -pthread )
+endif()
+
 # Add the unit tests directory
 add_subdirectory ( test )
 

